src/shared/TransactionBuilder.ts

```diff
@@ -77,7 +85,10 @@ export class TransactionBuilder {
   }
 
   getSigners(): Signer[] {
-    return this.records.flatMap((record) => record.signers);
+    const feePayer = this.feePayer ? [this.feePayer] : [];
```

Comment:
That's intentional. If the `feePayer` is not truthy then we don't want to add it as a signer. All the provided examples output the expected behaviour IMO.

Reply:
I know but aren't you checking for existence here really given that it is either [`Signer|undefined`](https://github.com/metaplex-foundation/js-next/blob/b7b9ff19bff5671d49958c105058099bd559a124/src/shared/TransactionBuilder.ts#L23)?

I'm just advocating for the more explicit check so people without context will know that this is not a `boolean` value you're comparing here.  It's obvious in this case, but I'm just saying that it would be a good idea to use `==/!= null` for those checks consistently throughout the code to make the code easier to understand.

Reply:
I can see scenarios where explicitly comparing to `null` brings value but here, I actually find that line easier to read and understand that way.

That being said, I don't feel that strongly about it so I'm happy to "disagree and commit" if that's a big deal to you.